internal/app/app.go

```diff
@@ -21,21 +23,30 @@ func NewApp(storageInstance *storage.Storage, config *config.Config) *App {
 }
 
 func (a *App) Start() {
-	err := dump.FillFromStorage(a.storageInstance, a.config.FilePath)
+	db, err := repository.InitDatabase(a.config.DBPath)
 	if err != nil {
 		log.Fatal(err)
 	}
+	dbDNSTurn := true
+	if a.config.DBPath == "" {
+		err = dump.FillFromStorage(a.storageInstance, a.config.FilePath)
+		if err != nil {
+			log.Fatal(err)
+		}
+		dbDNSTurn = false
+	}
 
-	err = api.StartRestAPI(a.config.ServerAddr, a.config.BaseURL, a.config.LogLevel, a.storageInstance)
+	err = api.StartRestAPI(a.config.ServerAddr, a.config.BaseURL, a.config.LogLevel, db, dbDNSTurn, a.storageInstance)
 	if err != nil {
 		log.Fatal(err)
 	}
 }
 
 func (a *App) Stop() {
-	err := dump.Set(a.storageInstance, a.config.FilePath, a.config.BaseURL)
-	if err != nil {
-		log.Fatal(err)
+	if a.config.DBPath == "" {
```

Comment:
```suggestion
	if a.config.UseDatabase() {
```
так становится чуть более читаемо, а реализация скрыта